### PR TITLE
MIPS: Update disassembler stub

### PR DESCRIPTION
**Pull request by xsacha**
Any ideas for how I can make MIPSDisAsm work for me here?

### CODE REVIEW

**Comment by hrydgard** [comment]
Just write a for loop?

**Comment by xsacha** [comment]
I mean in terms of using it for mips32r2 instead of PSP mips.

**Comment by hrydgard** [comment]
Won't it be at least semi-compatible, if you just use it as-is? Good enough for debugging bringup at least?

File: Core/MIPS/JitCommon/JitCommon.cpp [hunk 1]
```diff
@@ -232,4 +232,14 @@ std::vector<std::string> DisassembleX86(const u8 *data, int size) {
 	return lines;
 }
 
+#elif defined(MIPS)
+
+std::vector<std::string> DisassembleMIPS(const u8 *data, int size) {
+	std::vector<std::string> lines;
+	// TODO: A MIPS disassembler for MIPS32r2+
+	// Utilise MIPSDisAsm(MIPSOpcode op, u32 pc, char *out, bool tabsToSpaces)
```
Comment (unknownbrackets) [inline Core/MIPS/JitCommon/JitCommon.cpp]: Personally, I think a disassembler that's missing a few things is a lot better than none at all.  But, if we want it to be accurate, maybe we handle them like this:

```c++
char temp[512];
if (!MIPS64OverrideDisAsm(op, pc, temp)) {
    // Okay, this isn't different from the PSP, so just use the same disasm.
    MIPSDisAsm(op, pc, temp);
}
lines.push_back(temp);
```

Where we would slowly build out `MIPS64OverrideDisAsm`.  Its job would be:
1. Output invalid for opcodes that MIPSDisAsm would think are vfpu/etc.
2. Output valid opcodes for things the PSP lacks.

Everything else would just be sent to the PSP func, which would work totally fine.

-[Unknown]

File: ffmpeg [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 8dd84dc1ea8bc70da1a345109383f62974a9c2fc
+Subproject commit dd6d6a216d5411dbd5bbe902c2d529904184fc9d
```
Comment (unknownbrackets) [inline ffmpeg]: This needs to be updated with the latest commit.

-[Unknown]

File: Core/MIPS/JitCommon/JitCommon.cpp (hunk 1 above)
Comment (xsacha) [inline Core/MIPS/JitCommon/JitCommon.cpp]: Sounds look a good idea. By the way, I have my MIPS machine all ready but just don't know where to start.

Comment (unknownbrackets) [inline Core/MIPS/JitCommon/JitCommon.cpp]: We never got the jit working with only branches implemented, right?

-[Unknown]

File: ffmpeg (hunk 2 above)
Comment (xsacha) [inline ffmpeg]: Yeah it was difficult to work out the issue and then the git moved forward and a lot of the code became outdated.

**Comment by hrydgard** [comment]
Any further interest in this @xsacha ? Otherwise please close.

**Pull request closed without merging**